hail/src/main/scala/is/hail/io/fs/FS.scala

```diff
@@ -60,6 +70,86 @@ object BGZipCompressionCodec extends CompressionCodec {
   def makeOutputStream(os: OutputStream): OutputStream = new BGzipOutputStream(os)
 }
 
+object FSUtil {
+  def runOnExit(f: Runnable): Unit = {
+    Runtime.getRuntime.addShutdownHook(new Thread(f))
+  }
```

Comment:
I think this is unused as you've defined this default on the `FS` trait. I'm fine putting it on the trait so this can be deleted.